src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -82,17 +82,50 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 		{
 			AlterObjectSchemaStmt *alterObjectSchemaStmt =
 				(AlterObjectSchemaStmt *) parseTree;
-			char **relationName = &(alterObjectSchemaStmt->relation->relname);
-			char **relationSchemaName = &(alterObjectSchemaStmt->relation->schemaname);
+			ObjectType objectType = alterObjectSchemaStmt->objectType;
 
-			/* prefix with schema name if it is not added already */
-			SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+			if (objectType == OBJECT_STATISTIC_EXT)
+			{
+				RangeVar *stat = makeRangeVarFromNameList(
+					(List *) alterObjectSchemaStmt->object);
+
+				/* set schema name and append shard id */
+				SetSchemaNameIfNotExist(&stat->schemaname, schemaName);
```

Comment:
I think what we do here is not correct. `schemaName` is the schema name of the relation and we set the statistics' schema name to it. They don't have to be the same.
I think the reason we don't hit any errors here is that we already always qualify the `stat->schemaname` before.

Reply:
Yes, for now, it never comes here without a schema name. I added that for possible future needs

Reply:
Ok but if in the future we need to add schema name here (if it comes here without a schema name), we would be adding the wrong schema name. The `schemaName` parameter here is for the relation but we put it in statistics' schema name.

Reply:
makes sense, removing